### ISSUE TITLE
Alt-L (Find in Selection) replaces block selection with multiline selection

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#45587]
Does this issue occur when all extensions are disabled?: Yes
    VS Code Version: 1.89.1
    OS Version: Ubuntu Linux 22.04

1. Close the find/replace window if it's open (click on the X in the upper-right) 
2. Use Alt-Shift-Leftclick to make a block selection
3. Press Ctrl-H to start search
4. Press Alt-L to enable "Find in Selection"

The block selection is replaced with a multi-line selection, but only the lines that had text under the block are part of the multiline.

Before pressing Alt-L:
![image](https://github-com/microsoft/vscode/assets/1065860/535b1071-daf2-4a85-af99-e74d202d0196)

After pressing Alt-L:
![image](https://github-com/microsoft/vscode/assets/1065860/9e4c49f5-173c-4511-a7c9-3e1a99f4a189)

Weirdly, it appears that the find/replace does still honor the original block selection (it only replaces text in the original block), but the block itself is no longer visible.

Basically, it just looks like the block selection features was added recently and not thoroughly tested with find/replace.
